tests/test_api_lib_policy.py

```diff
@@ -370,9 +370,31 @@ def test_04a_check_max_active_token_user(self):
                         "type": "totp"}
         self.assertTrue(check_max_token_user(req))
 
+        # Now, we disable the token NEW001, so the user again has NO active token
+        enable_token("NEW001", enable=False)
+        # we enroll a new HOTP token, this would now succeed
+        init_token({"serial": "NEW002", "type": "hotp",
+                    "otpkey": "1234567890123456"},
+                   user=User(login="cornelius",
+                             realm=self.realm1))
+        tokenobject_list = get_tokens(user=User(login="cornelius",
+                                                realm=self.realm1))
+        self.assertTrue(len(tokenobject_list) == 2)
+        # now we enable the first hotp token again, which fails due to the policy
```

Comment:
This is not a valid test for enabling the token. 
Enabling the token is simply:
  
~~~~http
POST /token/enable
serial=<serial>
~~~~

So there is no user object in the request. However, you have a user in lines 384 and 385. Your line 698 in lib/prepolicy.py will not be covered.

Reply:
I removed the user data to match the enabling request.

Reply:
Well, I would have thought it would be great, if we actually had a real API test.
I manually checked the API and it was oK: